[
    {
        "content": "<p>Hi everyone, JupyterHub appears down for Casper right now.   Just checking if anyone is able to start JupyterLab on Casper DAV.</p>",
        "id": 3859,
        "sender_full_name": "Brian Bonnlander",
        "timestamp": 1584627957
    },
    {
        "content": "<p>it works for me.</p>",
        "id": 3860,
        "sender_full_name": "Anna-Lena Deppenmeier",
        "timestamp": 1584628252
    },
    {
        "content": "<p>Thanks! Are you using ssh-tunneling?</p>",
        "id": 3861,
        "sender_full_name": "Brian Bonnlander",
        "timestamp": 1584628283
    },
    {
        "content": "<p>yes</p>",
        "id": 3862,
        "sender_full_name": "Anna-Lena Deppenmeier",
        "timestamp": 1584628305
    },
    {
        "content": "<p>using execdav and then start-jupyter</p>",
        "id": 3863,
        "sender_full_name": "Anna-Lena Deppenmeier",
        "timestamp": 1584628351
    },
    {
        "content": "<p>I cannot connect to the browser when I ssh-tunnel.   do not have \"start-jupyter\".   Instead I am running:</p>\n<div class=\"codehilite\"><pre><span></span>(base) casper01:~$ execdav --account=STDD0003\nSubmitting interactive job to slurm ...\nSetting display for X forwarding ...\n\nRunning command:\nsalloc --time 6:00:00 --partition dav --ntasks 1  --account=STDD0003 srun --pty --mem 0  --export=PATH=/bin:/usr/bin,SHELL,TERM,HOME,DISPLAY,XAUTHORITY /bin/bash -l\n\nsalloc: Pending job allocation 4590150\nsalloc: job 4590150 queued and waiting for resources\nsalloc: job 4590150 has been allocated resources\nsalloc: Granted job allocation 4590150\nsalloc: Waiting for resource configuration\nsalloc: Nodes casper10 are ready for job\nResetting modules to system default\n(base) casper10:~$ jupyter lab --no-browser --ip=$(hostname)\n[I 08:30:38.006 LabApp] [nb_conda_kernels] enabled, 7 kernels found\n[I 08:30:40.904 LabApp] The port 8888 is already in use, trying another port.\n[I 08:30:42.764 LabApp] [jupyter_nbextensions_configurator] enabled 0.4.1\n[I 08:30:42.873 LabApp] JupyterLab extension loaded from /glade/u/home/bonnland/miniconda3/lib/python3.7/site-packages/jupyterlab\n[I 08:30:42.873 LabApp] JupyterLab application directory is /glade/u/home/bonnland/miniconda3/share/jupyter/lab\n[I 08:30:42.875 LabApp] Serving notebooks from local directory: /glade/u/home/bonnland\n[I 08:30:42.875 LabApp] The Jupyter Notebook is running at:\n[I 08:30:42.875 LabApp] http://casper10:8889/?token=dade741a8c9b67a8310943df559170eb36eb4531027876e5\n[I 08:30:42.875 LabApp]  or http://127.0.0.1:8889/?token=dade741a8c9b67a8310943df559170eb36eb4531027876e5\n[I 08:30:42.875 LabApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).\n[C 08:30:42.884 LabApp]\n\n    To access the notebook, open this file in a browser:\n        file:///glade/u/home/bonnland/.local/share/jupyter/runtime/nbserver-194264-open.html\n    Or copy and paste one of these URLs:\n        http://casper10:8889/?token=dade741a8c9b67a8310943df559170eb36eb4531027876e5\n     or http://127.0.0.1:8889/?token=dade741a8c9b67a8310943df559170eb36eb4531027876e5\n</pre></div>\n\n\n<p>Then I am giving this command in another local window:  </p>\n<div class=\"codehilite\"><pre><span></span>ssh -N -L 8889:casper10:8889 casper10.ucar.edu\n</pre></div>\n\n\n<p>Does this look correct?   Should my browser then be able to reach \"<a href=\"http://casper10:8889/?token=dade741a8c9b67a8310943df559170eb36eb4531027876e5\" target=\"_blank\" title=\"http://casper10:8889/?token=dade741a8c9b67a8310943df559170eb36eb4531027876e5\">http://casper10:8889/?token=dade741a8c9b67a8310943df559170eb36eb4531027876e5</a>\" ?</p>",
        "id": 3864,
        "sender_full_name": "Brian Bonnlander",
        "timestamp": 1584628791
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"9\">@Brian Bonnlander</span> you should go to <a href=\"http://localhost:8889\" target=\"_blank\" title=\"http://localhost:8889\">http://localhost:8889</a></p>",
        "id": 3865,
        "sender_full_name": "Michael Levy",
        "timestamp": 1584628836
    },
    {
        "content": "<p>(including the token string is fine)</p>",
        "id": 3866,
        "sender_full_name": "Michael Levy",
        "timestamp": 1584628855
    },
    {
        "content": "<p>Thank you everyone, I try....</p>",
        "id": 3867,
        "sender_full_name": "Brian Bonnlander",
        "timestamp": 1584628948
    },
    {
        "content": "<p>If that doesn't work, try changing the <code>casper10.ucar.edu</code> to <code>casper01.ucar.edu</code> in your <code>ssh -N -L</code> step.</p>",
        "id": 3868,
        "sender_full_name": "Michael Levy",
        "timestamp": 1584628983
    },
    {
        "content": "<p>I was thinking that the casper compute nodes are only accessible through the login nodes, but if you can ssh directly to <code>casper10.ucar.edu</code> then that is not the case</p>",
        "id": 3869,
        "sender_full_name": "Michael Levy",
        "timestamp": 1584629040
    },
    {
        "content": "<p>Your suggestion of <a href=\"http://localhost:8889\" target=\"_blank\" title=\"http://localhost:8889\">http://localhost:8889</a> works for me.   Thank you again.</p>",
        "id": 3871,
        "sender_full_name": "Brian Bonnlander",
        "timestamp": 1584629126
    },
    {
        "content": "<p>One more question (hopefully):  My ssh-tunnel session is working, but my Dask dashboard URL is apparently not working.  It assigned a random port to me, but it seems problematic that \"<a href=\"http://jupyterhub.ucar.edu/dav\" target=\"_blank\" title=\"http://jupyterhub.ucar.edu/dav\">jupyterhub.ucar.edu/dav</a>\" is part of the URL.   If <a href=\"http://jupyterhub.ucar.edu/dav\" target=\"_blank\" title=\"http://jupyterhub.ucar.edu/dav\">http://jupyterhub.ucar.edu/dav</a> is not resolving in my browser, should I expect my Dask dashboard URL to work?   The full URL is \"<a href=\"https://jupyterhub.ucar.edu/dav/user/bonnland/proxy/44116/status\" target=\"_blank\" title=\"https://jupyterhub.ucar.edu/dav/user/bonnland/proxy/44116/status\">https://jupyterhub.ucar.edu/dav/user/bonnland/proxy/44116/status</a>\".</p>",
        "id": 3873,
        "sender_full_name": "Brian Bonnlander",
        "timestamp": 1584630396
    },
    {
        "content": "<p>Use <code>proxy/44116</code> in the dask lab extension</p>",
        "id": 3874,
        "sender_full_name": "Deepak Cherian",
        "timestamp": 1584630966
    },
    {
        "content": "<p>Thank you!  That works.</p>",
        "id": 3875,
        "sender_full_name": "Brian Bonnlander",
        "timestamp": 1584631053
    },
    {
        "content": "<p>Currently my attempts to run execdav are not working.   Has anyone else had trouble with this in the past?   Does this happen if you exceed your core hour allocation?   This same command was working for me before.</p>\n<div class=\"codehilite\"><pre><span></span><span class=\"o\">(</span>base<span class=\"o\">)</span> casper26:~$ execdav --account<span class=\"o\">=</span>STDD0003\nSubmitting interactive job to slurm ...\nNo display<span class=\"p\">;</span> X forwarding disabled ...\n\nRunning command:\nsalloc --time <span class=\"m\">6</span>:00:00 --partition dav --ntasks <span class=\"m\">1</span>  --account<span class=\"o\">=</span>STDD0003 srun --pty --mem <span class=\"m\">0</span>  --export<span class=\"o\">=</span><span class=\"nv\">PATH</span><span class=\"o\">=</span>/bin:/usr/bin,SHELL,TERM,HOME /bin/bash -l\n\nsalloc: error: Job submit/allocate failed: Invalid account or account/partition combination specified\n</pre></div>",
        "id": 5384,
        "sender_full_name": "Brian Bonnlander",
        "timestamp": 1585336064
    },
    {
        "content": "<p>I see now that I forgot to ssh with X11 forwarding, but I still get the same error when using \"ssh -Y\" to login.</p>",
        "id": 5386,
        "sender_full_name": "Brian Bonnlander",
        "timestamp": 1585336533
    },
    {
        "content": "<p>OK, I finally worked my way around <a href=\"http://sam.ucar.edu\" target=\"_blank\" title=\"http://sam.ucar.edu\">sam.ucar.edu</a> and discovered that my core hour usage has gone over quota.  So this is probably the source of the error message.</p>",
        "id": 5390,
        "sender_full_name": "Brian Bonnlander",
        "timestamp": 1585337503
    }
]